### PR TITLE
Profiling and Optimisation 🚀

### PR DESCRIPTION
**Pull request by jcblemai**
## Docs
Introduce variable `COVID_LOGLEVEL`, by default its value is `INFO`.

Level | Numeric value | Comment
-- | -- | --
CRITICAL | 50 | 
ERROR | 40 | skip most checks, display time for full call only: **USE IN PRODUCTION**
WARNING | 30 |
**INFO** | 20 | (DEFAULT) Print time elapsed for most function, performs sanity check: **USE FOR DRY RUNS** 
DEBUG | 10 | show lot of stuff such as parameter value and reduction. Very verbose.
NOTSET | 0

## TODO:
- Separate SEIR and NPI, put utils and profiling tools in NPI
- Remove useless checks
- Use fast_setup
- remove profiling hooks from minimal interface with a envvar

## DONE:
- Optimise Reduce to with smart indexing of geoids
- Optimise MutltiTimeReduce to use group structure wittily indexing
- Time function in production
- Change printing to logging
- Setup envar for logging
- In depth profiling of  SEIR + Outcome OK
    - File prefixed `init` a very large portion of the code is spend building interventions
- clean output from R and python
- Raise Error when the reduction is not written to file (instead of warning)

Cleanup:

### CODE REVIEW

**Comment by jcblemai** [comment]
```
  INFO:root:[onerun_SEIR.NPI] completed in 171.5189826488495 s
  INFO:root:[onerun_SEIR.seeding] completed in 1.5924654006958008 s
  INFO:root:[onerun_SEIR.pdraw] completed in 0.17375659942626953 s
  INFO:root:[onerun_SEIR.reduce] completed in 0.568389892578125 s
  INFO:root:[onerun_SEIR.compute] completed in 68.44127416610718 s
  INFO:root:[onerun_SEIR.postprocess] completed in 8.238770723342896 s
  INFO:root:[onerun_SEIR] completed in 256.5899569988251 s
  INFO:root:[onerun_delayframe_outcomes.compute] completed in 25.671432733535767 s
  INFO:root:[onerun_delayframe_outcomes.postprocess] completed in 1.3761711120605469 s
  INFO:root:[onerun_OUTCOMES] completed in 56.321040868759155 s
```

**Comment by jcblemai** [comment]
```
INFO:root:[onerun_SEIR.NPI] completed in 58.29847431182861 s
INFO:root:[onerun_SEIR.seeding] completed in 1.3667593002319336 s
INFO:root:[onerun_SEIR.pdraw] completed in 0.16362333297729492 s
INFO:root:[onerun_SEIR.reduce] completed in 0.5950396060943604 s
INFO:root:[onerun_SEIR.compute] completed in 63.649850368499756 s
INFO:root:[onerun_SEIR.postprocess] completed in 7.572793245315552 s
INFO:root:[onerun_SEIR] completed in 131.83866262435913 s
INFO:root:[onerun_delayframe_outcomes.compute] completed in 25.346678495407104 s
INFO:root:[onerun_delayframe_outcomes.postprocess] completed in 1.3493804931640625 s
```

**Comment by jcblemai** [comment]
## After MultiTimeReduce Optimisation

Screenshot 2021-04-23 at 14 50 00

File: Outcomes/outcomes.py [hunk 1]
```diff
@@ -62,18 +66,19 @@ def onerun_delayframe_outcomes_load_hpar(config, in_run_id, in_prefix, in_sim_id
...
-            global_config=npi_config[1],
-            geoids=places,
-            loaded_df = npi_load(
-                file_paths.create_file_name_without_extension(
-                    in_run_id,
-                    in_prefix, 
-                    in_sim_id,
-                    "hnpi"),'parquet')
-        )
+        with Timer('onerun_delayframe_outcomes_load_hpar.NPI'):
```
Comment (jkamins7) [inline Outcomes/outcomes.py:69]: You're leaving in the Timer calls?

Comment (jcblemai) [reply Outcomes/outcomes.py:69]: yes, they're only print something depending on logging level, and don't take a lot of time, if that's ok

**Pull request merged**